index.xml

```diff
@@ -374,6 +376,41 @@ tampered with in transit.
    </t>
 
   </section>
+
+  <section anchor="date-header" title="Date Header Usage">
+
+   <t>
+This specification makes use of the Date header in illustrations,
+in the default list of headers to sign where none is explicitly provided, and
+as a one of headers to include as a minimum in the signature in implementations.
+   </t>
+
+   <t>
+The meaning and construction of the Date header is the one defined in
+<xref target="RFC7231">RFC 7231</xref>, <eref
+target="https://tools.ietf.org/html/rfc7231#section-7.1.1.2">
+Section 7.1.1.2</eref>.
+Note that the definition used defines the Date header as required in all
+Responses, but OPTIONAL in Requests unless it will
+"convey useful information to the server".
+   </t>
+
+   <t>
+Where a signature incorporates the Date header in a Request
+- by default or explicitly - it is
+assumed clients are capable of including a properly formatted Date
+representation in all Requests. It is therefore a requirement that all
```

Comment:
This has tended to make the IoT people cranky -- requiring IoT devices to include a proper date... so we need to have an easy mechanism that enables IoT devices to not include a date. Seems like `headers="foo-header"` (where it omits date) might be that mechanism?

Reply:
@msporny where do IoT people specify the date then?

Reply:
In some cases, they don't. They don't want to specify the Date because their devices have no way of fixing clock skew, so Date is unreliable. Instead, they usually sign a message counter or something else that the IoT hub tracks.

Reply:
So essentially moving `date` as a recommended `Signature` field instead of a mandatory header would:

  - work for IoT
  - work for Requests

Reply:
When building a solution I was too lazy to actually read the standard and present a formatted message, so instead just made an OPTIONS/HEAD request to my target - as Date is mandatory in all responses so I now have the date, and specifically exactly the date format the target understands and stated is the date. So, this is not a showstopper for IoT if done right.

That said, what we're verifying is that the date header as stated is signed, but not what it contains. It could be a literal ``now``, seconds since boot or in the Hebrew or Mayan calendars for all we (this specification) cares. It's already a minimum *recommended* header, and the default if the parameter is absent, but implementations are free to ignore it and specify any other headers they want.

Reply:
> So essentially moving date as a recommended Signature field instead of a mandatory header...

It's not mandatory now, but it is required in the default case. A good implementation should be aware of the client's ability to compose and sign the required headers and balance this with their security requirements e.g. Data and what it means. In https://github.com/w3c-dvcg/http-signatures/pull/47 we even mention signing no headers at all (so signing the empty string), which is pretty useless, but technically permissible.

Reply:
@msporny 
> Seems like `headers="foo-header"` (where it omits date) might be that mechanism?

That's exactly what I was trying to convey - if you ask for ``date``, make sure your clients can actually send one, but if not then no worries... A good workaround is a challenge-response, and in a way the "fetch date from server" approach is on that theme, but that's not for us (this spec) to say. If this doesn't come across clear enough in the wording let me know.

Reply:
@ioggstream Does this resolve your concerns or should I add specific wording to clarify anything you're not happy with?

Reply:
Hi @liamdennehy sorry for the late reply... The statement clarifies the usage, but I think it's a "fixing by warning" ;)

Reply:
I think that's about all we can do given the breadth of (in)capabilities for clients. Resolve or needs more work?
For example, we can add verbiage on the contents not necessarily being the actual time, and the meaning of the content is up to the implementer to interpret, but we're already getting a bit verbose.